citesphere/src/main/webapp/WEB-INF/views/auth/group/editItem.jsp

```diff
@@ -140,6 +180,31 @@ $(function() {
 	    }, 1000);
 	});
 	
+	$(".author-item").on("click", function(){
+		var authorItem = $(this);
+		$("#firstNameAuthor").val(authorItem.attr("data-author-firstname"));
+		$("#lastNameAuthor").val(authorItem.attr("data-author-lastname"));
+		$("#uriAuthor").val(authorItem.attr("data-author-uri"));
+		$("#idAuthor").val(authorItem.attr("data-author-id"));
+		
+		for(var i=0;i<authorItem.children().length-1;i++){
+			var item = authorItem.children().eq(i);
+			var affInput = $("#affiliationTemplate").clone();
+			affInput.removeAttr("id");
+			affInput.find("input").attr("data-affiliation-name", item.data("affiliationName"));
+			affInput.find("input").attr("data-affiliation-id", item.data("affiliationId"));
+			affInput.find("input").val(item.data("affiliationName"));
+			$("#affiliations").append(affInput);
+		}
+		if(authorItem.children().length-1>0){
+			$("#affiliationTemplate").hide();
+		}
+		if($("#updateAuthorButton").length == 0){
```

Comment:
when would that be the case? I think I know what you're doing but I'm pretty sure that is a more elegant way. Why don't you just reuse the existing code and just change the label of the button from 'Add Author' to 'Update Author' when a user clicks on the edit button?